src/app/(admin)/new_dashboard/hooks/useContacts.ts

```diff
@@ -1,22 +1,44 @@
-import { useState } from "react";
-import { Contact } from "../types";
+import { useEffect, useState } from "react";
+import { Chat } from "@/types";
+import { useChats } from "@/hooks/api/chatHooks";
+import { SuccessResponse } from "@/api/base.client";
+import socket from "@/socket";
 
 export const useContacts = () => {
-  const initialContacts: Contact[] = [
-    { name: "Alice", avatar: "/avatar1.png", lastMessage: "See you later!" },
-    {
-      name: "Bob",
-      avatar: "/avatar2.png",
-      lastMessage: "Let's catch up tomorrow.",
-    },
-    {
-      name: "Charlie",
-      avatar: "/avatar3.png",
-      lastMessage: "Can you send me the file?",
-    },
-  ];
+  const [contacts, setContacts] = useState<Chat[]>([]);
+  const [selectedChat, setSelectedChat] = useState<Chat | null>(null);
 
-  const [contacts, setContacts] = useState<Contact[]>(initialContacts);
+  const { isError, data, error } = useChats();
 
-  return { contacts, setContacts };
+  useEffect(() => {
+    if (data) {
+      const { data: chats } = data as SuccessResponse<Chat[]>;
+      setContacts(chats);
+      setSelectedChat(chats[0]);
+    }
+  }, [data]);
+
+  if (isError) {
+    console.error(error);
+  }
```

Comment:
```suggestion
  const { data: chatData, isError, error } = useChats({
    onSuccess: (data) => {
      const { data: chats } = data;
      setSelectedChat((prev) => prev ?? chats[0]); 
    },
  });
```

Why not set the data with tanstack query?


Also return the 'isError' and you use it to display some sort of error

Reply:
![image](https://github.com/user-attachments/assets/c3e6fced-97c0-4697-beb2-f90c1fe2e4a7)

Reply:
Great point, I didn't know that. 

I recommend doing something like this, having a custom generic hook similar to onSuccess so we could use it in other parts aswell.

Examples:

https://dev.to/barrymichaeldoyle/tanstacks-react-query-kicked-onsuccess-onerror-and-onsettled-out-of-usequery-now-what-2i33


`import { useEffect } from "react";

/**
 * Custom hook to handle onSuccess logic similar to the deprecated `onSuccess` in useQuery.
 * @param {any} data - The data to watch for changes.
 * @param {boolean} isSuccess - Indicates if the query was successful.
 * @param {Function} onSuccess - The callback function to run when data is successfully fetched.
 */
const useOnSuccess = (data, isSuccess, onSuccess) => {
  useEffect(() => {
    if (isSuccess && data) {
      onSuccess(data);
    }
  }, [data, isSuccess, onSuccess]);
};

export default useOnSuccess;

  // Use custom onSuccess hook to set chat messages when data is successfully fetched
  useOnSuccess(data, isSuccess, (fetchedData) => {
    const { data: chat } = fetchedData as SuccessResponse<Chat>;
    const messages = chat.messages as Message[];
    setChatMessages(messages);
  });
`